src/GitHubExtension/DeveloperId/LoginUIController.cs

```diff
@@ -1,18 +1,31 @@
 ﻿// Copyright (c) Microsoft Corporation and Contributors
 // Licensed under the MIT license.
 
-using Microsoft.Windows.ApplicationModel.Resources;
+using System.Net;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using GitHubExtension.Client;
+using GitHubExtension.DeveloperId.LoginUI;
+using GitHubExtension.Helpers;
 using Microsoft.Windows.DevHome.SDK;
 using Windows.Foundation;
 
 namespace GitHubExtension.DeveloperId;
-internal class LoginUIController : IExtensionAdaptiveCardSession
+public class LoginUIController : IExtensionAdaptiveCardSession
 {
+    private readonly IDeveloperIdProviderInternal _developerIdProvider;
     private IExtensionAdaptiveCard? _loginUI;
-    private static readonly LoginUITemplate _loginUITemplate = new ();
+    private Uri? _hostAddress;
 
-    public LoginUIController()
+    public Uri HostAddress
     {
+        get => _hostAddress ?? throw new InvalidOperationException("HostAddress is null");

```

Comment:
The user has to set `HostAddress` before using it?  Consider adding comments explaining this.